actor/builtin/paymentbroker/paymentbroker_test.go

```diff
@@ -195,6 +195,114 @@ func TestPaymentBrokerRedeemWithCondition(t *testing.T) {
+		sys := setup(t)
+		require.NoError(t, sys.st.SetActor(context.TODO(), toAddress, actor.NewActor(pbTestActorCid, types.NewZeroAttoFIL())))
+
+		// Expect that the conditions are nil on init
+		paymentBroker := state.MustGetActor(sys.st, address.PaymentBrokerAddress)
+		channel := sys.retrieveChannel(paymentBroker)
+		assert.Nil(t, channel.Conditions)
+
+		// Successfully redeem the payment channel
+		condition := &types.Predicate{To: toAddress, Method: method, Params: payerParams}
+		appResult, err := sys.applySignatureMessage(sys.target, 100, types.NewBlockHeight(0), 0, "redeem", 0, condition, redeemerParams...)
+		require.NoError(t, err)
+		require.NoError(t, appResult.ExecutionError)
+
+		// Expect that the conditions are now set and correct
+		paymentBroker = state.MustGetActor(sys.st, address.PaymentBrokerAddress)
+		channel = sys.retrieveChannel(paymentBroker)
+		assert.NotNil(t, channel.Conditions)
+		assert.Equal(t, toAddress, channel.Conditions.To)
+		assert.Equal(t, method, channel.Conditions.Method)
+		assert.Contains(t, channel.Conditions.Params, addrParam.Bytes())
+		assert.Contains(t, channel.Conditions.Params, sectorIdParam)
+		assert.Contains(t, channel.Conditions.Params, blockHeightParam.Bytes())
+	})
+
+	t.Run("Redeem should set cached conditions back to nil when no condition is provided", func(t *testing.T) {
+		sys := setup(t)
+		require.NoError(t, sys.st.SetActor(context.TODO(), toAddress, actor.NewActor(pbTestActorCid, types.NewZeroAttoFIL())))
+
+		// Successfully redeem the payment channel with conditions
+		condition := &types.Predicate{To: toAddress, Method: method, Params: payerParams}
+		appResult, err := sys.applySignatureMessage(sys.target, 100, types.NewBlockHeight(0), 0, "redeem", 0, condition, redeemerParams...)
+		require.NoError(t, err)
+		require.NoError(t, appResult.ExecutionError)
+
+		// Expect that the conditions are set and correct
+		paymentBroker := state.MustGetActor(sys.st, address.PaymentBrokerAddress)
+		channel := sys.retrieveChannel(paymentBroker)
+		assert.NotNil(t, channel.Conditions)
+		assert.Equal(t, toAddress, channel.Conditions.To)
+		assert.Equal(t, method, channel.Conditions.Method)
+		assert.Contains(t, channel.Conditions.Params, addrParam.Bytes())
+		assert.Contains(t, channel.Conditions.Params, sectorIdParam)
+		assert.Contains(t, channel.Conditions.Params, blockHeightParam.Bytes())
+
+		// Successfully redeem the payment channel again without conditions
+		appResult, err = sys.applySignatureMessage(sys.target, 200, types.NewBlockHeight(0), 0, "redeem", 0, nil, redeemerParams...)
+		require.NoError(t, err)
+		require.NoError(t, appResult.ExecutionError)
+
+		// Expect that the conditions are now nil
+		paymentBroker = state.MustGetActor(sys.st, address.PaymentBrokerAddress)
+		channel = sys.retrieveChannel(paymentBroker)
+		assert.Nil(t, channel.Conditions)
+	})
+
+	t.Run("Redeem uses cached conditions in subsequent calls", func(t *testing.T) {
```

Comment:
For documentation's sake, it would be good to run the 'without params' case before running the redeem with params to establish that it fails, then run it after to show it succeeds.

Also we need an executes cached condition test for the the `Close` method.